test/e2e/pod.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Pods", func() {
+	var (
+		c   *client.Client
+		err error
+	)
+
+	BeforeEach(func() {
+		c = loadClientOrDie()
+	})
+
+	It("should submit and remove a pod", func() {
+		podClient := c.Pods(api.NamespaceDefault)
+
+		By("loading the pod json")
+		pod := loadPodOrDie(assetPath("api", "examples", "pod.json"))
+		value := strconv.Itoa(time.Now().Nanosecond())
+		pod.Name = pod.Name + "-" + randomSuffix()
+		pod.Labels["time"] = value
+		pod.Spec.Containers[0].Ports[0].HostPort = 0
+
+		By("submitting the pod to kubernetes")
+		_, err = podClient.Create(pod)
+		if err != nil {
+			Fail(fmt.Sprintf("Failed to create pod: %v", err))
+		}
+		defer podClient.Delete(pod.Name)
```

Comment:
The should update a pod test below also calls defer GinkgoRecover() inside of this defer. The two specs should be consistent.